### PR TITLE
Feat: Add self updating cache action

### PR DESCRIPTION
**Pull request by nicolassanmar**
Task: https://www.notion.so/Open-Source-pnpm-CI-Actions-63a0f2b09c994a898b877482a246f2d6

Add a reusable self updating cache action, that uses `actions/cache` and allows it to update the cache for the next job that requires it

### CODE REVIEW

File: self-updating-cache/action.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name: "Self Updating Cache"
+description: "Keeps and updated cache leveraging actions/cache and the recipe for using updated caches (https://github.com/actions/cache/blob/main/tips-and-workarounds.md#update-a-cache)"
```
Comment (cau777) [inline self-updating-cache/action.yml]: Nit: I think you meant either "Keeps an updated" or "Keeps and updates"

File: self-updating-cache/action.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        lookup-only: ${{ inputs.lookup-only }}
+        save-always: ${{ inputs.save-always }}
+    - name: Set output
+      id: cache-hit
+      shell: bash
+      run: |
+        echo "cache-hit=${{ steps.cache-hit.outputs.cache-hit }}" >> $GITHUB_OUTPUT
+branding:
+  icon: "archive"
+  color: "gray-dark"
```
Comment (cau777) [inline self-updating-cache/action.yml:66]: Question: What is "branding" used for?

**Review by cau777 — APPROVED** [review]
Merge when ready!
Just an idea: What do you think of keeping a list of existing actions in the main README.md, to serve as a project overview?

File: self-updating-cache/action.yml (hunk 2 above)
Comment (nicolassanmar) [reply self-updating-cache/action.yml:66]: It provides a way to define an icon on https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#branding.
This is the same icon that `actions/cache` has. Maybe we can change the color later (if we even appear on the marketplace, as I'm not sure that monorepos are listed there).

Added a comment about it

File: self-updating-cache/action.yml (hunk 1 above)
Comment (nicolassanmar) [reply self-updating-cache/action.yml]: Thanks! I did mean "Keeps an updated"

**Comment by nicolassanmar** [comment]
> Merge when ready! Just an idea: What do you think of keeping a list of existing actions in the main README.md, to serve as a project overview?

Agreed! I'd like to make this have some autogenerated docs based on the `.yaml` files, as otherwise, it's pretty likely that the README.md will get outdated eventually.
I'll work on that after I push all the actions

**Pull request merged**